### PR TITLE
Move append_vec::test_utils into its own file

### PR DESCRIPTION
**Pull request by brooksprumo**
See title 😊

### CODE REVIEW

File: runtime/src/append_vec.rs [hunk 1]
```diff
@@ -28,6 +28,8 @@ use {
     },
 };
 
+pub mod test_utils;
```
Comment (jeffwashington) [inline runtime/src/append_vec.rs:31]: I'm not too familiar with the right way to do this code organization.
Does the pub mod go here or should it be in lib?

Comment (brooksprumo) [reply runtime/src/append_vec.rs:31]: Putting it here preserves the current usage. So it's namespaced like this (both before this PR, and with this PR):
```
runtime::append_vec::test_utils
```

If this line were in `runtime/lib.rs` then the namespacing would change to:
```
runtime::test_utils
```
which I don't think is right.

Comment (jeffwashington) [reply runtime/src/append_vec.rs:31]: not a big deal at any rate

**Review by jeffwashington — APPROVED** [review]
lgtm

**Pull request merged**